content/en/contribute/code/core/dev-environment.md

```diff
@@ -18,25 +18,42 @@ These steps apply to both 3.x and 4.x CHT core development, unless stated otherw
 
 ## The Happy Path Installation
 
-This CHT Core developer guide will have you install NodeJS, npm, Grunt and CouchDB (via Docker) on your local workstation. These instructions should work verbatim on Ubuntu 18-22 (see [Ubuntu 18 note](#ubuntu-1804)), but will need tweaks for MacOS (via `brew`, see [MacOS > 12.3 note](#macos--123)) or Windows (via WSL2).
+This CHT Core developer guide will have you install NodeJS, npm, Grunt and CouchDB (via Docker) on your local workstation.
 
 ### Install NodeJS, npm, grunt and Docker
 
-First, update your current Ubuntu packages and install some supporting tools via `apt`:
+First, update your current packages and install some supporting tools:
 
-```shell
+{{< tabpane persistLang=false lang=shell >}}
+{{< tab header="Linux (Ubuntu)" >}}
 sudo apt update && sudo apt -y dist-upgrade
 sudo apt -y install xsltproc curl uidmap jq python2 git make g++
-```
-
-Then install `nvm`, add it to your path and install NodeJS 16:
-
-```shell
+# Use NVM to install NodeJS:
 export nvm_version=`curl -s https://api.github.com/repos/nvm-sh/nvm/releases/latest | jq -r .name`
-curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/$nvm_version/install.sh | $0
-. ~/.$0rc
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/$nvm_version/install.sh | $SHELL
+. ~/.$(basename $SHELL)rc
```

Comment:
Is there a reason to switch from `$0` to `$(basename $SHELL)`?  I didn't know for sure either way so tested it, which looks like `$0` is better (see video below).  As well [this SO](https://stackoverflow.com/questions/60844165/definitively-determine-if-currently-running-shell-is-bash-or-zsh) post suggests `$0` is best.

https://user-images.githubusercontent.com/8253488/224393495-b6ed87bd-6e1f-4994-820f-9449e59bf721.mp4

Reply:
Yeah, as I noted in the PR description, `$0` did not work at all for me on WSL2 or on Linux with zsh.  (Though `basename $0` would have been okay on Linux...).  I get that using the _default_ shell vs the _current_ shell is perhaps not ideal, but IMHO if it is true that `$SHELL` has more consistent behavior across platforms, I think it is worth it...